test/parallel/test-net-better-error-messages-port-hostname.js

```diff
@@ -3,12 +3,18 @@ const common = require('../common');
 const net = require('net');
 const assert = require('assert');
 
-const c = net.createConnection(common.PORT, '***');
+const s = net.createServer();
+s.listen(0);
+const port = s.address().port;
```

Comment:
I think we are overthinking here. As the port doesn't matter for this test, we can simply do the following and be done with it.

```diff
diff --git a/test/parallel/test-net-better-error-messages-port-hostname.js b/test/parallel/test-net-better-error-messages-port-hostname.js
index 14f84eb15b..896fe29b38 100644
--- a/test/parallel/test-net-better-error-messages-port-hostname.js
+++ b/test/parallel/test-net-better-error-messages-port-hostname.js
@@ -3,12 +3,12 @@ const common = require('../common');
 const net = require('net');
 const assert = require('assert');

-const c = net.createConnection(common.PORT, '***');
+const c = net.createConnection(0, '***');

 c.on('connect', common.mustNotCall());

 c.on('error', common.mustCall(function(e) {
   assert.strictEqual(e.code, 'ENOTFOUND');
-  assert.strictEqual(e.port, common.PORT);
+  assert.strictEqual(e.port, 0);
   assert.strictEqual(e.hostname, '***');
 }));
```